layers/sctp.go

```diff
@@ -304,19 +298,25 @@ func decodeSCTPData(data []byte, p gopacket.PacketBuilder) error {
 		StreamSequence:  binary.BigEndian.Uint16(data[10:12]),
 		PayloadProtocol: SCTPPayloadProtocol(binary.BigEndian.Uint32(data[12:16])),
 	}
+
+	if sc.PayloadProtocol == 0 {
+		sc.Payload = make([]byte, sc.SCTPChunk.Length-16)
+		copy(sc.Payload, data[16:sc.SCTPChunk.Length+1])
+		// Length is the length in bytes of the data, INCLUDING the 16-byte header.
+		p.AddLayer(sc)
```

Comment:
Make sure you call p.SetApplicationLayer(sc) here, to set the ApplicationLayer() pointer to the first application layer seen in the packet.

Reply:
I think,this isn't the right place for `p.SetApplicationLayer(sc)`. The payload that is contained inside the SCTPData chunk isn't decoded. And since it can be anything, there will be no decoder for it. The layer for the protocol is set at a different place https://github.com/google/gopacket/blob/master/layers/sctp.go#L34